score.cpp

```diff
@@ -0,0 +1,124 @@
+#include <iostream>
+#include "score.h"
+
+
+Score::Score(){
+    //bestScore = 0;
+    currentScore = 0;
+    setScoreFile();
+    uploadBestScore();
+}
+
+Score::Score(std::string specialPath){
+    //bestScore = 0;
+    currentScore = 0;
+
+
+    std::filesystem::path spcPath = specialPath;
+    if(std::filesystem::exists(spcPath)){
+        std::filesystem::create_directories(spcPath / "data");
+        pathToFile = spcPath / "data/scores.txt";
+        std::cout << "Path to score file: " << pathToFile << std::endl;
+    }else{
+        setScoreFile();
+    }
+
+    uploadBestScore();
+}
+
+void Score::setScoreFile(){
+    try{
+        std::filesystem::path pathToRun = std::filesystem::current_path();
+        //std::cout << "Path to project: " << pathToProject << std::endl;
+
+        while(pathToRun != pathToRun.root_path()){ //Looking for project directory (it will be mistake if CMakeLists.txt is deleted)
+            std::cout << "Current path: " << pathToRun << std::endl;
+            if(std::filesystem::exists(pathToRun / "CMakeLists.txt")){
+
+                std::filesystem::path pathToProject = pathToRun;
+                std::cout << "Path to project: " << pathToProject << std::endl;
+
+                std::filesystem::path pathToFloder = pathToProject / "data";
+                std::filesystem::create_directories(pathToFloder);
+
+                pathToFile = pathToFloder / "scores.txt";
+                std::cout << "Path to score file: " << pathToFile << std::endl;
+                break;
+            }
+            pathToRun = pathToRun.parent_path();
+        }
+    }
+    catch(const std::filesystem::filesystem_error &err){
+        std::cerr << "File processing error: " << err.what() << std::endl;
+    }
+}
+
+void Score::saveCurrentScoreToFile(){
+    if(currentScore != 0){
+        scoreFile.open(pathToFile, std::ios::app);
+        if(!scoreFile.is_open()){
+            std::cerr << "File not open and score not saved" << std::endl;
+            return;
+        }
+        std::cout << "File is open..." << std::endl;
+
+
+        scoreFile << "," << currentScore;
```

Comment:
The format of your score should not be ,score,score,score.
Do it at least from new line and you should store the scores in JSON format